src/standalone/Vegetation/Canopy.jl

```diff
@@ -483,9 +483,10 @@ function ClimaLSM.make_update_aux(
                     )
                 ) / 2
         end
-        @. β = moisture_stress(ψ[n_stem + n_leaf] * ρ_l * grav, sc, pc)
-        # We update the fa[n_stem+n_leaf] element once we have computed transpiration, below
+        i_end = n_stem + n_leaf
```

Comment:
likewise here, and Im not totally clear on the rules around broadcasting -> use $$ but this seemed to give what we want!

Reply:
Same as above, I think it's fine to add an intermediate if it helps fix this issue.

Reply:
Do we need to use two `$$` if we have a broadcast inside of a loop because the broadcast and loop are each an expression, so we need to interpolate the index value twice? If we have a broadcast outside of a loop, do we use just one `$`?

Reply:
If we broadcast outside of a loop with a known index (like 2), we dont need any dollar signs:
 ```
julia> @. ψ.:2 = PlantHydraulics.water_retention_curve(
               retention_model,
               PlantHydraulics.effective_saturation(ν, ϑ_l.:2),
               ν,
               S_s,
           )
Float64-valued Field:
  [203.938]
```

If we want to index at the result of an expression, it doesnt really work. I think we need to define the intermediate i_end = n_stem+n_leaf.
```

julia> @. ψ.:$(n_stem+n_leaf) = PlantHydraulics.water_retention_curve(
               retention_model,
               PlantHydraulics.effective_saturation(ν, ϑ_l.:$(n_stem+n_leaf)),
               ν,
               S_s,
           )
ERROR: syntax: "(n_stem + n_leaf)" is not a valid function argument name around REPL[118]:1
```